src/components/ble/NotificationManager.cpp

```diff
@@ -68,44 +68,50 @@ NotificationManager::Notification NotificationManager::GetPrevious(NotificationM
     result = *(notifications.end() - 1);
   else
     result = *(currentIterator - 1);
-  
+
   if (result.id >= id)
-    return Notification {};
+    return {};
 
   result.index = (lastNotification.id - result.id) + 1;
   return result;
 }
 
 void NotificationManager::Dismiss(NotificationManager::Notification::Id id) {
-  if(empty)
+  if (empty)
     return;
 
-  bool found = false;
-  size_t foundIndex;
-  size_t count = NbNotifications();
-  for(size_t i = 0; i < TotalNbNotifications; i++) {
-    if(notifications[i].id == id) {
-      foundIndex = i;
-      found = true;
-      break;
-    } 
-  }
-  if(!found)
+  auto currentIterator = std::find_if(notifications.begin(), notifications.end(), [id](const Notification& n) {
+    return n.valid && n.id == id;
+  });
+  if (currentIterator == notifications.end() || currentIterator->id != id)
     return;
 
-  for(size_t i = foundIndex; i < TotalNbNotifications - 1; i++) {
-    notifications[i] = notifications[i+1];
+  size_t count = NbNotifications();
+  size_t foundIndex = std::distance(notifications.begin(), currentIterator);
+  for (size_t i = foundIndex; i < TotalNbNotifications - 1; i++) {
+    notifications[i] = notifications[i + 1];
     notifications[i].index = i;
   }
+  notifications[TotalNbNotifications - 1] = {};
+
+  // fill any gaps in ids
+  for (size_t i = 0; i < TotalNbNotifications; i++) {
+    if (!notifications[i].valid)
+      break;
```

Comment:
If it can happen, that the first element is not valid, then the id-reassignment will do nothing. I think we should use `continue` instead of `break` to make sure we go through all the messages, no matter where the ring-buffer is filled currently

```suggestion
    if (!notifications[i].valid)
      continue;
```

Reply:
Take a look at the latest implementation in 9fedd9bf1b3ed1109fdb269b8d2c615617de8bc9
This logic no longer exists, instead the loop is checking any id's > dismissed notification id.

Good eye though, thanks!